server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -949,6 +953,13 @@ public int hashCode() {
         boolean supportsElementType(ElementType elementType) {
             return elementType != ElementType.BYTE;
         }
+
+        @Override
+        boolean updatableTo(IndexOptions update) {
+            return update.type.equals(this.type)
+                || update.type.equals(VectorIndexType.HNSW.name)
```

Comment:
Would it be better to check for non-flat types? I guess we will be adding int4 / binary any time soon, so we would have to modify this again?

Reply:
thanks Tommaso! Makes sense to leave as is for now